.github/workflows/pr.yaml

```
@@ -0,0 +1,278 @@
+    - name: Create tool_test_output.html
+      run: 'planemo test_reports tool_test_output.json --test_output tool_test_output.html'
+    - name: Copy artifacts into place
+      run: mkdir upload && mv tool_test_output.json tool_test_output.html upload
+    - uses: actions/upload-artifact@v1
+      with:
+        name: 'Tool test output ${{ matrix.chunk  }}'
+        path: upload
+
+  combine_outputs:
+    name: Combine chunked test results
+    needs: test
+    # This job runs on Linux
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/download-artifact@v1
+        with:
+          name: Tool test output 0
+      - uses: actions/download-artifact@v1
+        with:
+          name: Tool test output 1
+      - uses: actions/download-artifact@v1
+        with:
+          name: Tool test output 2
+      - uses: actions/download-artifact@v1
+        with:
+          name: Tool test output 3
+      - uses: actions/setup-python@v1
+        with:
+          python-version: '3.7'
+          architecture: 'x64'
+      - name: setup planemo
+        run: 'pip install planemo'
+      - name: setup jq
+        run: 'sudo apt-get install jq'
+      - name: combine outputs
+        run: |
+          find . -name tool_test_output.json -exec sh -c 'planemo merge_test_reports "$@" tool_test_output.json' sh {} +
+      - name: Create tool_test_output.html
+        run: 'planemo test_reports tool_test_output.json --test_output tool_test_output.html'
+      - name: Copy artifacts into place
+        run: mkdir upload && mv tool_test_output.json tool_test_output.html upload
+      - uses: actions/upload-artifact@v1
+        with:
+          name: 'All tool test results'
+          path: upload
+      - name: check status of combined status
+        run: |
+             stat=$(jq '.["tests"][]["data"]["status"]' upload/tool_test_output.json | grep -v "success" || true)
+             if [[ "$(echo -n $stat | wc -l)" -gt "0" ]]; then
+                 echo "unsuccessful tests found $(sort <<<$stat | uniq -c)"
+                 exit 1
+             fi
+
+  deploy:
+    name: Deploy
+    needs: combine_outputs
+    # This job runs on Linux
+    runs-on: ubuntu-latest
+    if: github.ref == 'refs/heads/master'
+    steps:
+    - uses: actions/setup-python@v1
+      with:
+        python-version: ${{ matrix.python-version }}
+        architecture: 'x64' # (x64 or x86)
+    - name: Cache .cache/pip
+      uses: actions/cache@v1
+      id: cache-pip
+      with:
+        path: ~/.cache/pip
+        key: pip_cache_${{ matrix.python-version }}_${{ env.GALAXY_RELEASE }}
+    - name: setup planemo
+      run: 'pip install planemo'
+    - name: planemo ci_find_repos
+      run: |
+           mkdir changed_repositories
+           planemo ci_find_repos --changed_in_commit_range origin/master... --exclude packages --exclude deprecated --output changed_repositories.list
+    - name: deploy on testtoolshed
```

```suggestion
    - name: deploy on testtoolshed
        env:
          SHED_EMAIL: ${{ secrets.shed_email }}
          SHED_PASSWORD: ${{ secrets.shed_password }}
```
This is described in https://help.github.com/en/actions/configuring-and-managing-workflows/creating-and-storing-encrypted-secrets.